packages/core/src/Keymap.ts

```diff
@@ -0,0 +1,101 @@
+import { CommandIdentifier, CommandArgs } from './Dispatcher';
+
+interface ShortcutPattern {
+    key: string;
+    modifiers: Set<string>;
+}
+// todo: should I place this in file Command?
+export interface CommandCall {
+    commandId: string;
+    commandArgs?: CommandArgs;
+}
+
+interface ShortcutItem extends CommandCall {
+    pattern: ShortcutPattern;
+    priority: number;
+}
```

Comment:
I'm not a fan of the two interfaces above. They are suspiciously similar to the `Shortcut` interface in `JWPlugin` that I expected to find here.

Reply:
The difference lies in the pattern that is not a `string` but a `ShortuctPattern`.
I think too much extends is not necessary. Here is the alternative:
```typescript
interface CommandCall {
    commandId: string;
    commandArgs?: CommandArgs;
}

interface ShortcutItemPriority extends CommandCall {
    priority: number;
}

interface ShortcutItem extends ShortcutItemPriority {
    pattern: ShortcutPattern;
}

interface Shortcut extends ShortcutItemPriority {
    pattern: string;
}
```
Do you prefer this version?